components/omega/doc/design/VertAdv.md

```diff
@@ -0,0 +1,347 @@
+(omega-design-vert-adv)=
+# Vertical Advection
+
+## 1 Overview
+
+The Vertical Advection module is responsible for the vertical transport of mass,
+momentum, and tracers within Omega. This implementation is consistent with
+Omega's non-Boussinesq governing equations, where the prognostic vertical
+coordinate is a pseudo-height, effectively a normalized pressure coordinate. The
+methods compute the tendencies associated with the vertical advection terms in
+the governing equations, and are designed to allow for flexibility in numerical
+scheme configuration.
+
+## 2 Requirements
+
+### 2.1 Requirement: Compute transport pseudo-velocity
+
+Compute the transport pseudo-velocity through vertical layer interfaces based on
+the divergence of the horizontal velocity field. Allow for possible movement of
+layer interfaces and tilt of layer interfaces
+
+### 2.2 Requirement: Advection scheme flexibility
+
+Support for multiple possible reconstruction schemes, allowing users to select
+among different orders of accuracy and upwinding at runtime through
+configuration options. Standard and monotonic flux-corrected transport (FCT)
+advection algorithms are available for the tendency calculations.
+
+### 2.3 Requirement: Compute tendencies for mass, velocity, and tracers
+
+Separate methods are needed to compute the vertical-advection tendencies
+for mass, velocity, and tracers. Mass (`LayerThickness`) and horizontal
+velocity (`NormalVelocity`) are stored as 2-D arrays. `LayerThickness` is
+cell-centered, while `NormalVelocity` is defined on edges. These tendencies
+use a second-order flux formulation that ensures strict conservation and energy
+consistency with the continuity equation. `Tracers` is a 3-D, cell-centered
+array. The tracer tendencies are computed using a configurable flux-form scheme
+that supports multiple interface reconstruction options.
+
+### 2.4 Requirement: Modular design
+
+The vertical advection module must consist of classes and methods that are easily
+incorporated into different time-integration schemes and maintain compatibility
+with the Omega driver framework.
+
+### 2.5 Desired: Monotonicity checks
+
+The vertical advection module should include optional monotonicity checks to
+ensure that reconstructed tracer fields remain physically bounded and free of
+spurious oscillations when using the FCT scheme.
+
+## 3 Algorithmic Formulation
+
+The [Omega V1 Governing Equations](omega-design-governing-eqns-omega1) describe
+the evolution of mass, velocity, and tracers, including the contributions
+from vertical advection.
+
+### 3.1 Pseudo-velocity
+
+The vertical pseudo-velocity is derived from the divergence of the horizontal
+velocity $d\tilde{w} / d\tilde{z} = - \nabla \cdot {\bf u}.$ In
+discrete form, the pseudo-velocity at the top interface of layer $k$ in cell
+$i$ is computed from the thickness-weighted divergence of horizontal velocity:
+
+$$
+\tilde{w}_{i,k-1/2} = \tilde{w}_{i,k+1/2} +
```

Comment:
Since we are planning on using the "p-star" coordinate, should we explicitly define $$\tilde{W}$$ to include the ALE term?  That would mean adding the pseudo-height/velocity equivalent of equations (4) and (5) from [Petersen et al. 2015](https://doi.org/10.1016/j.ocemod.2014.12.004).